AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/ResHandler.java

```diff
@@ -91,6 +102,10 @@ public void process(Element element, EComponentHolder holder) {
 				methodBody.assign(ref(fieldName), classes().HTML.staticInvoke("fromHtml").arg(invoke(holder.getResourcesRef(), resourceMethodName).arg(idRef)));
 			} else if (element.getAnnotation(DrawableRes.class) != null && hasContextCompatInClasspath()) {
 				methodBody.assign(ref(fieldName), classes().CONTEXT_COMPAT.staticInvoke("getDrawable").arg(holder.getContextRef()).arg(idRef));
+			} else if (element.getAnnotation(DrawableRes.class) != null && hasContextWithGetDrawableMethod() && !hasContextCompatInClasspath()) {
```

Comment:
Maybe if you just create an if `(element.getAnnotation(DrawableRes.class) != null)` then call a method and handle all `@DrawableRes` related things, it would be cleaner and better structured.